.github/scripts/ensure_actions_will_cancel.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import yaml
+
+from pathlib import Path
+
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+WORKFLOWS = REPO_ROOT / ".github" / "workflows"
+
+
+def concurrency_key(filename):
+    workflow_name = filename.with_suffix("").name.replace("_", "-")
+    return f"{workflow_name}-${{{{ github.event.pull_request.number || github.sha }}}}"
+
+
+SKIP = {
+    "update_disabled_tests.yml",
+    "stale_pull_requests.yml",
+    "update_s3_htmls.yml",
+    "push_nightly_docker_ghcr.yml",
+}
+
+
+def should_skip(filename):
+    if filename.name in SKIP:
+        return True
+
+    with open(filename, "r") as f:
+        content = f.read()
+        if "@generated DO NOT EDIT MANUALLY" in content:
+            return True
+
+    data = yaml.safe_load(content)
+    on = str(data.get("on", data.get(True, None)))
+    if "'pull_request'" not in on and "'workflow_dispatch'" not in on:
```

Comment:
Probably a dumb question but why do these have doubly-nested quotes? Is it looking at the raw string data rather than the parsed YAML? And if so, why?

Reply:
It was checking the `str(entry)` rather than `entry` for some weird reason that I forgot, it's fixed now with the changes from the comments above